### ISSUE TITLE
Tool not works with mogoose models when model and collection names are different

### ISSUE THREAD

**Issue by vologab** [balmasi/migrate-mongoose#21]
Migration tool not works when  Mongoose model defined with 3 arguments - `mongoose.model('model', ModelSchema, 'model-collection');`. It  uses model collection `model` instead `model-collection`,  when call `this('model')`

Example code:
```
require('../model1');
require('../model2');

const bucketBefore = 'aaa';
const bucketAfter = 'bbb';

function updateImages(Model, bucketBefore, bucketAfter, done, imgField = 'img') {
    let modelCursor = Model.find({}).cursor();

    modelCursor.eachAsync(doc => {
        if (doc[imgField] && doc[imgField].includes(bucketBefore)) {
            doc[imgField] = doc[imgField].replace(bucketBefore, bucketAfter);
            doc.save();
        }
    }, done);

    modelCursor.on('error', err => done('migration error: ' + err));
}

exports.up = function up (done) {
    // works fine for such kind of models -  mongoose.model('model1', ModelSchema1);
    // but not works for models with different collection name -
    //  mongoose.model('model2', ModelSchema2, 'model2-collection');

    updateImages(this('model1'), bucketBefore, bucketAfter, done, 'image'); // <- OK
    updateImages(this('model2'), bucketBefore, bucketAfter, done, 'image'); // <- Problem

};

exports.down = function down(done) {
// ...
};

```

**Comment by balmasi** [comment]
@vologab I suggest not using this('model2') and `require` ing directly into your migration. I used to think it was a good idea to make this('model') available to the migrator, but I don't think the migrator should know about any models, let alone registering them on a separate mongoose connection. Your code should be handling that because all a migration framework should do is run migrations.

What do you think? @vologab

**Comment by vologab** [comment]
@balmasi  But if I will create separate connection, how I can obtain connection settings from migrator config?

**Comment by balmasi** [comment]
@vologab mmm good question.

My question in response is, why is that necessary? 

You're assuming everyone wants their models to exist on the same db as their migrations db.

I think the config should only be to tell the migrator what DB/collection to use to WRITE and READ  migration states, not the actual model.

Wouldn't you have your connection settings from the environment your code runs in? If running migrations in prod, your models should already be registered using the prod connection in your code. 

Please let me know if you think I'm missing something

**Comment by balmasi** [comment]
Closing this because I think I've allowed enough time for discussion.

TLDR; I think the migrator should open an independent connection to the database rather than making assumptions about the database configurations of the user
